maat-court-data-api/src/main/java/gov/uk/courtdata/dces/service/ConcorContributionsService.java

```diff
@@ -0,0 +1,99 @@
+import gov.uk.courtdata.entity.ContributionFilesEntity;
+import gov.uk.courtdata.exception.MAATCourtDataException;
+import gov.uk.courtdata.repository.ConcorContributionsRepository;
+import gov.uk.courtdata.repository.ContributionFilesRepository;
+import gov.uk.courtdata.util.ValidationUtils;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ConcorContributionsService {
+    private static final String DB_USER_NAME = "MLA";
+    private final ConcorContributionsRepository concorRepository;
+    private final ContributionFilesRepository contributionFileRepository;
+    private final ContributionFileMapper contributionFileMapper;
+
+    public List<String> getConcorFiles(final ConcorContributionStatus status) {
+        log.info("Getting concor contribution file with status with the -> {}", status);
+        final List<ConcorContributionsEntity> concorFileList = concorRepository.findByStatus(status);
+        return concorFileList.stream().map(ConcorContributionsEntity::getCurrentXml).toList();
+    }
+
+    @Transactional(rollbackFor = MAATCourtDataException.class)
+    public boolean createContributionAndUpdateConcorStatus(final ConcorContributionRequest contributionRequest){
+
+        ValidationUtils.isNull(contributionRequest,"contributionRequest object is null");
+        ValidationUtils.isEmpty(contributionRequest.getContributionIds(),"ContributionIds are empty/null.");
+
+        final ContributionFilesEntity contributionFilesEntity = createContributionFile(contributionRequest);
+        return updateConcorStatusForContribution(contributionRequest.getContributionIds(), SENT, contributionFilesEntity.getId());
+    }
+
+    private ContributionFilesEntity createContributionFile(final ConcorContributionRequest contributionRequest) {
+
+        final ContributionFilesEntity contributionFileEntity;
+        try {
+            log.info("Updating the concor contribution file ref  -> {}", contributionRequest);
+            contributionFileEntity = contributionFileMapper.toContributionFileEntity(contributionRequest);
+            // when file is not provided then xml generate.
+            contributionFileEntity.setFileName(Optional.ofNullable(contributionRequest.getXmlFileName()).orElse(getFilename()));
+            contributionFileEntity.setUserCreated(DB_USER_NAME);
+            contributionFileEntity.setUserModified(DB_USER_NAME);
+
+            contributionFileRepository.save(contributionFileEntity);
+        } catch (Exception e) {
+            throw new MAATCourtDataException("ConcorContributionRequest to ContributionEntity Mapper error.");
```

Comment:
I think this exception message could be misleading, as an exception could be thrown by a failure to persist the entity, while the mapping from `ConcorContributionRequest -> ContributionFilesEntity` was successful.

You should consider including exception message in the `MAATCourtDataException` raised.

e.g.
```
String message = "Failed to map ConcorContributionRequest to ContributionFilesEntity and persist: [%s]".formatted(e.getMessage());
throw new MAATCourtDataException(message);
```